libsolidity/parsing/Parser.cpp

```diff
@@ -118,12 +118,15 @@ void Parser::parsePragmaVersion(SourceLocation const& _location, vector<Token> c
 	static SemVerVersion const currentVersion{string(VersionString)};
 	// FIXME: only match for major version incompatibility
 	if (!matchExpression.matches(currentVersion))
-		m_errorReporter.fatalParserError(
-			_location,
-			"Source file requires different compiler version (current compiler is " +
-			string(VersionString) + " - note that nightly builds are considered to be "
-			"strictly less than the released version"
-		);
+		// If m_parserErrorRecovery is true, the same message will appear from SyntaxChecker::visit(),
```

Comment:
Ok this is weird - it somehow sounds like we really want a fatal error here...
On the other hand, if we only do that if recovery is requested, then it's probably fine.

Reply:
I'm not convinced parser errors can be recovered from, in most of the cases.

Reply:
@axic Ok - what do you recommend?

Reply:
Here are two other alternatives that I can think of. Perhaps there are more...

* turn fatal error when error recovery is requested into a non-fatal error and live with the same message appearing twice
* make the error recovery stop unconditionally after ast generation.